src/main/java/com/frcteam364/CTREModuleState.java

```diff
@@ -0,0 +1,56 @@
+package com.frcteam364;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.SwerveModuleState;
+
+public class CTREModuleState {
+
+  /**
+   * Minimize the change in heading the desired swerve module state would require by potentially
+   * reversing the direction the wheel spins. Customized from WPILib's version to include placing
+   * in appropriate scope for CTRE onboard control.
+   *
+   * @param desiredState The desired state.
+   * @param currentAngle The current module angle.
+   */
+  public static SwerveModuleState optimize(SwerveModuleState desiredState, Rotation2d currentAngle) {
+    double targetAngle = placeInAppropriate0To360Scope(currentAngle.getDegrees(), desiredState.angle.getDegrees());
+    double targetSpeed = desiredState.speedMetersPerSecond;
+    double delta = targetAngle - currentAngle.getDegrees();
+    if (Math.abs(delta) > 90){
+        targetSpeed = -targetSpeed;
+        targetAngle = delta > 90 ? (targetAngle -= 180) : (targetAngle += 180);
+    }        
+    return new SwerveModuleState(targetSpeed, Rotation2d.fromDegrees(targetAngle));
+  }
+
+  /**
+     * @param scopeReference Current Angle
+     * @param newAngle Target Angle
+     * @return Closest angle within scope
+     */
```

Comment:
Can you add a description to the doc comment of what this is doing? It looks like it does some math to prevent jumping from 360 to 0 (since they are the same in real world) to prevent unneeded rotation? Is this correct?

This is my assumption, but it would be nice to have that written down for the end user or if/when we need to refactor or review code.

Reply:
Updated the doc comment to this.

"Takes the current angle and the target angle and outputs a new target angle that is within +/- 180 degrees of the current angle. The difference of the current angle of the module and the desired angle could be greater than 360 degrees, and that's a problem, and this function solves it. Note that this is simply helping the optimize function. This function only gets the angle within +/- 180 degrees, NOT +/- 90 degrees which is what the optimize function does."